homeassistant/components/sensor/date_countdown.py

```diff
@@ -0,0 +1,69 @@
+"""
+Date Countdown
+For more details about this sensor please refer to the documentation at
+https://home-assistant.io/components/sensor.date_countdown/
+
+"""
+
+import datetime
+import logging
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import CONF_NAME
+from homeassistant.helpers.entity import Entity
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = "DateCountdown"
+CONF_DATE = 'date'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_DATE): cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up certificate expiry sensor."""
+    end_date = config.get(CONF_DATE)
+    sensor_name = config.get(CONF_NAME)
+
+    add_devices([Countdown(sensor_name, end_date)])
+
+
+class Countdown(Entity):
+    """Implementation of the date countdown sensor."""
+
+    def __init__(self, sensor_name, end_date):
+        """Initialize the sensor."""
+        self.end_date = end_date
+        self._name = sensor_name
+        self._state = None
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def icon(self):
+        """Icon to use in the frontend, if any."""
+        return 'mdi:calendar'
+
+    def update(self):
+        end_date = datetime.datetime.strptime(self.end_date, '%m-%d-%Y %H:%M')
+        days = (end_date - datetime.datetime.now())
+
+        days, seconds = days.days, days.seconds
+        hours = seconds // 3600
+        minutes = (seconds % 3600) // 60
+        seconds = seconds % 60
+        self._state = str(days) + " days " + str(hours) + " hours " + str(minutes) + " minutes"
```

Comment:
I would like to suggest that the state is a single number. Perhaps hours (in an addition step we could make it configurable, days, hours, etc.). This would make it possible to use the value for automation rules.

To keep the details (days, hours, minutes, and seconds) make them available as `device_state_attributes`. Again, could be useful for automation.